### PR TITLE
Prepare changelog and upgrade guide for v4.0.0

### PR DESCRIPTION
**Pull request by ralphtheninja**

### CODE REVIEW

**Comment by ralphtheninja** [comment]
Got some error from Sauce Labs:

```
2018-12-22T16:50:39.267Z airtap:sauce:android:6.0 quitting browser
2018-12-22T16:50:39.267Z airtap:sauce:MicrosoftEdge:18 quitting browser
2018-12-22T16:50:39.267Z airtap:sauce:firefox:64 shutdown
2018-12-22T- failed: <android 6.0 on Linux>, (0, 0)
/home/travis/build/Level/[secure]/node_modules/airtap/bin/airtap.js:255
          throw err
          ^
Error: android@6.0: [init({"build":"1237","name":"[secure]","browserName":"android","version":"6.0","platform":"Linux","tunnel-identifier":"1237.4"})] The environment you requested was unavailable.
Sauce could not start your job. For more information on what happened, please visit https://saucelabs.com/jobs/04b83ccdb5134d119c6455b565fc7d81: {"status": 13, "sessionId": "04b83ccdb5134d119c6455b565fc7d81", "value": {"message": "Sauce could not start your job. For more information on what happened, please visit https://saucelabs.com/jobs/04b83ccdb5134d119c6455b565fc7d81"}}
    at /home/travis/build/Level/[secure]/node_modules/wd/lib/webdriver.js:140:15
    at Request._callback (/home/travis/build/Level/[secure]/node_modules/wd/lib/http-utils.js:89:7)
    at Request.self.callback (/home/travis/build/Level/[secure]/node_modules/request/request.js:186:22)
    at Request.emit (events.js:189:13)
    at Request.<anonymous> (/home/travis/build/Level/[secure]/node_modules/request/request.js:1163:10)
    at Request.emit (events.js:189:13)
    at IncomingMessage.<anonymous> (/home/travis/build/Level/[secure]/node_modules/request/request.js:1085:12)
    at Object.onceWrapper (events.js:277:13)
    at IncomingMessage.emit (events.js:194:15)
    at endReadableNT (_stream_readable.js:1129:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! [secure]@3.1.1 test-browsers: `airtap --loopback airtap.local test/index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the [secure]@3.1.1 test-browsers script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2018-12-22T16_50_46_370Z-debug.log
```

Will restart build to see if it's just temporary.

**Comment by ralphtheninja** [comment]
screenshot from 2018-12-22 18-01-13

**Comment by ralphtheninja** [comment]
Temporary hickup.

File: UPGRADING.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 This document describes breaking changes and how to upgrade. For a complete list of changes including minor and patch releases, please refer to the [changelog](CHANGELOG.md).
 
+## v4
+
+There has been many major updates to dependencies but the main reason for this major is due to `level-iterator-stream` which now is based on `readable-stream@3` which has dropped support for `IE10`.
```
Comment (vweevers) [inline UPGRADING.md]: There are more reasons, more important than dropping IE10 (btw we also dropped IE10 in `abstract-leveldown`).

```suggestion
There have been two major updates to dependencies. First, `level-iterator-stream` is now based on [`readable-stream@3`](https://github.com/nodejs/readable-stream#version-3xx). Second, `deferred-leveldown` is now based on [`abstract-leveldown@6`](https://github.com/Level/abstract-leveldown/blob/master/UPGRADING.md#v6). Please follow these links for more information; both contain significant enough changes to warrant this `levelup` major. In addition, all aforementioned dependencies and by extension `levelup` have dropped support of IE10. 

To get a consistent behavior between opening and opened `levelup` instances (in the former case, your store will be wrapped with `deferred-leveldown`), we recommend to pair `levelup@4` only with a store based on `abstract-leveldown` >= 6. For example, `deferred-leveldown` now rejects `null` and `undefined` values. If you pair `levelup@4` with an older store, `db.put('key', null)` would only throw an error if `db` is still opening itself.
```

Comment (ralphtheninja) [reply UPGRADING.md]: :+1: I was counting on you to fill in the blanks

**Pull request merged**